src/Core/Resolvers/Factories/MutationEngineFactory.cs

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Net;
+using Azure.DataApiBuilder.Auth;
+using Azure.DataApiBuilder.Config.ObjectModel;
+using Azure.DataApiBuilder.Core.Configurations;
+using Azure.DataApiBuilder.Core.Models;
+using Azure.DataApiBuilder.Core.Services.MetadataProviders;
+using Azure.DataApiBuilder.Service.Exceptions;
+using Microsoft.AspNetCore.Http;
+
+namespace Azure.DataApiBuilder.Core.Resolvers.Factories
+{
+    /// <summary>
+    /// MutationEngineFactory class.
+    /// Used to get the IMutationEngine based on database type.
+    /// </summary>
+    public class MutationEngineFactory : IMutationEngineFactory
+    {
+        private readonly IEnumerable<IMutationEngine> _mutationEngines;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MutationEngineFactory"/> class.
+        /// </summary>
+        /// <param name="runtimeConfigProvider">runtimeConfigProvider.</param>
+        /// <param name="queryManagerFactory">queryManagerFactory</param>
+        /// <param name="metadataProviderFactory">metadataProviderFactory.</param>
+        /// <param name="cosmosClientProvider">cosmosClientProvider</param>
+        /// <param name="queryEngineFactory">queryEngineFactory.</param>
+        /// <param name="httpContextAccessor">httpContextAccessor.</param>
+        /// <param name="authorizationResolver">authorizationResolver.</param>
+        /// <param name="gQLFilterParser">GqlFilterParser.</param>
+        public MutationEngineFactory(RuntimeConfigProvider runtimeConfigProvider,
+            IQueryManagerFactory queryManagerFactory,
+            IMetadataProviderFactory metadataProviderFactory,
+            CosmosClientProvider cosmosClientProvider,
+            IQueryEngineFactory queryEngineFactory,
+            IHttpContextAccessor httpContextAccessor,
+            IAuthorizationResolver authorizationResolver,
+            GQLFilterParser gQLFilterParser)
+        {
+            _mutationEngines = new List<IMutationEngine>();
+
+            bool sqlEngineNeeded = runtimeConfigProvider.GetConfig().GetDataSourceNamesToDataSourcesIterator().Any
+                (x => x.Value.DatabaseType == DatabaseType.MSSQL || x.Value.DatabaseType == DatabaseType.PostgreSQL || x.Value.DatabaseType == DatabaseType.MySQL);
```

Comment:
Can we instead use `x.Value.DatabaseType != DatabaseType.CosmosDB_NoSQL`?

Reply:
Thats not very safe in case you introduce a new database type. Then it could wrongly create the sqlEngine.